ipv8/attestation/trustchain/database.py

```diff
@@ -97,6 +97,22 @@ def get_all_blocks(self):
         """
         return self._getall(u"", ())
 
+    def get_number_of_known_blocks(self, public_key=None):
+        """
+        Return the total number of blocks in the database or the number of known blocks for a specific user.
+        """
+        if public_key:
+            return list(self.execute(u"SELECT COUNT(*) FROM blocks WHERE public_key = ?",
+                                     (database_blob(public_key), )))[0][0]
+        return list(self.execute(u"SELECT COUNT(*) FROM blocks"))[0][0]
+
+    def remove_recent_blocks(self, num_blocks_to_remove):
+        """
+        Remove recent blocks from the database.
```

Comment:
Hmm I'm not sure why I named this method like this since it actually removes oldest blocks. And yes, it might remove blocks of your own chain, good catch 👍